scripts/build/Jenkins/Jenkinsfile

```diff
@@ -483,8 +483,10 @@ def ExecuteControlFile(controlFile, repoName, processed_repos, o3de_project, def
     //queue enable all gems specified in the control file
     if(controlFile.ENABLE_GEMS) {
         controlFile.ENABLE_GEMS.each { gemName ->
-            defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
-        }
+            if(o3de_project) {
+                defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
+            }
+        }    
```

Comment:
Should we place the check for the `o3de_project` variable outside of the foreach loop.

There is no reason to iterate over the gem names if the project name isn't set at all

Reply:
I think putting it outside of the loop would only check if the list is empty; the issue is that there may be still be elements with project names along with empty ones. We could use a `-` operator to remove all null and blank elements, though. Will try that in sandbox

Reply:
Unfortunately, it didn't make a difference. It also ran twice, even after removing the null elements. It's as though list length is being retained somehow. May be a quirk of Java/groovy that needs further investigation